src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/resthandler/RestRetryFailedManagedIndex.kt

```diff
@@ -0,0 +1,168 @@
+            populateList(state, startState)
+
+            val builder = channel.newBuilder().startObject()
+            if (listOfIndexMetadata.isNotEmpty()) {
+                val updateManagedIndexMetaDataRequest = UpdateManagedIndexMetaDataRequest(listOfIndexMetadata)
+
+                try {
+                    client.execute(UpdateManagedIndexMetaDataAction, updateManagedIndexMetaDataRequest,
+                        object : RestResponseListener<AcknowledgedResponse>(channel) {
+                            override fun buildResponse(acknowledgedResponse: AcknowledgedResponse): RestResponse {
+                                if (acknowledgedResponse.isAcknowledged) {
+                                    builder.field(UPDATED_INDICES, listOfIndexMetadata.size)
+                                } else {
+                                    failedIndices.addAll(listOfIndexMetadata.map {
+                                        FailedIndex(it.first.name, it.first.uuid, "failed to update IndexMetadata")
+                                    })
+                                }
+                                buildInvalidIndexResponse(builder)
+                                return BytesRestResponse(RestStatus.OK, builder.endObject())
+                            }
+                        }
+                    )
+                } catch (e: ClusterBlockException) {
+                    failedIndices.addAll(listOfIndexMetadata.map {
+                        FailedIndex(it.first.name, it.first.uuid, "failed to update with ClusterBlockException. ${e.message}")
+                    })
+                    buildInvalidIndexResponse(builder)
+                    channel.sendResponse(BytesRestResponse(RestStatus.OK, builder.endObject()))
+                }
+            } else {
+                buildInvalidIndexResponse(builder)
+                channel.sendResponse(BytesRestResponse(RestStatus.OK, builder.endObject()))
+            }
+        }
+
+        private fun buildInvalidIndexResponse(builder: XContentBuilder) {
+            if (failedIndices.isNotEmpty()) {
+                builder.field(FAILURES, true)
+                builder.startArray(FAILED_INDICES)
+                for (failedIndex in failedIndices) {
+                    builder.startObject()
+                    builder.field("index_name", failedIndex.name)
+                    builder.field("index_uuid", failedIndex.uuid)
+                    builder.field("reason", failedIndex.reason)
+                    builder.endObject()
+                }
+                builder.endArray()
+            } else {
+                builder.field(FAILURES, false)
+            }
+        }
+
+        private fun populateList(state: ClusterState, startState: String?) {
+            for (indexMetadataEntry in state.metaData.indices) {
+                val indexMetadata = indexMetadataEntry.value
+
+                if (indexMetadata.getPolicyName() == null) {
```

Comment:
We also do not manage indices with empty `""` or blank `" "` policy names. We might have a helper extension function to validate policy name.

Reply:
`getPolicyName` will return null since it checks if the string is `isNullOrBlank`.